### PR TITLE
Print the data in hexa

### PR DESCRIPTION
**Pull request by PanPriape**
Sorry, i'm french and not confortable with English.

In order to add, in the "print" function, a visualisation in hexadecimal format.
For example : 

08  20  00 00 00 00 00 00 33 31 34 31 37 36 39 34 31 38 |......3141769418|
08  21  30 34 30 30 36 34 30 30 58 31 39 39 33 30 33 31 |04006400X1993031|
08  22  34 30 35 34 32 34 39 33 53 30 35 34 32 32 33 36 |40542493S0542236|
08  23  3c e2 0d 08 04 44 78 77 88 00 00 00 00 00 00 00 |<....Dxw........|

In the program it's looks better because everything fit well. There isn' a column greater than an another.

### CODE REVIEW

**Comment by 4ZM** [comment]
Hi! I'm on vacation right now. Will take a look at this in a couple of days. Cheers

**Comment by PanPriape** [comment]
Hi !

Happy holydays ;)

2016-07-12 9:01 GMT+02:00 Anders <notifications@github.com>:

File: util.c [hunk 1]
```diff
@@ -66,9 +66,101 @@ void print_hex_array(const unsigned char* data, size_t nbytes) {
...
+  for (i = 0; i < size; i++) {
+    if (inpout[i] == ' ') {
+      rep++;
+    }
+  }
+
+  return rep;
+}
+
+char *hexToAscii(char *inpout) {
```
Comment (PanPriape) [inline util.c]: Convert a string which represents hexadecimal in a string which represents an ASCII string

**Comment by 4ZM** [comment]
I've had a look at your code. Thanks for sharing it! I really like your idea of showing an ASCII representation of the data. However, I think that your implementation is unnecessarily complicated.

I've commited a reimplementation of your work to master, please have a look at it and let me know if it's missing something or if it doesn't do what you want:

https://github.com/4ZM/mfterm/commit/0d447453bb15887bb1f50dd4c66674fca5be7e33

**Comment by PanPriape** [comment]
It's exactly what i want, thank you.

I have an other modification in the mifare_ctrl.c file ( line 356 ). When i
lunch the make it said that it miss some "{ }".

I don't if it just for my computer.

Anyway thank you for your quick answers. The new functionnality is exactly
what i want to do. In a less complicated way ;).

Have a good day.

2016-07-15 18:49 GMT+02:00 Anders <notifications@github.com>:

**Comment by 4ZM** [comment]
Great!

About that mifare_ctrl.c file ( line 356 ). Send me the output of running make and I'll have a look at it.

**Comment by PanPriape** [comment]
I joined a screnn shot of the make.

The erreur is :
erreur : this ‘if’ clause does not guard... [-Werror=misleading-indentation]
     if (key_type == MF_KEY_UNLOCKED && is_trailer_block(block))
     ^~

When i add  "{}", the make finished whith no problem.


2016-07-18 16:12 GMT+02:00 Anders <notifications@github.com>:

**Comment by 4ZM** [comment]
Thanks a bunch. Now I see the problem, it's a bug. God catch of your compiler. Wish mine would have warned about this.  I committed your suggested change to master.

**Comment by PanPriape** [comment]
My pleasure :)

**Pull request closed without merging**